packages/astro/src/assets/vite-plugin-assets.ts

```diff
@@ -229,6 +230,11 @@ export default function assets({
 						});
 					}
 
+					if (/\.svg$/.test(id)) {
+						const { contents, ...metadata } = imageMetadata;
+						return makeSvgComponent(metadata, contents!);
```

Comment:
A comment explaining why `contents` is always defined would be great, you can also cast `imageMetadata` to avoid the `!` (still with a comment explaining why it gets casted)